service/history/workflow/update/registry.go

```diff
@@ -398,7 +398,8 @@ func (r *registry) checkTotalLimit() error {
 	if len(r.updates)+r.completedCount >= r.maxTotal() {
 		r.instrumentation.countTooMany()
 		return serviceerror.NewFailedPrecondition(
-			fmt.Sprintf("limit on number of total updates has been reached (%v)", r.maxTotal()),
+			fmt.Sprintf("The limit on the number of distinct updates in this workflow has been reached (%v). "+
```

Comment:
Suggest retaining "total". To my mind that helps convey that this isn't a limit on instantaneous # of concurrent updates or something like that.

```suggestion
			fmt.Sprintf("The limit on the total number of distinct updates in this workflow has been reached (%v). "+
```

Reply:
Sounds good!